### ISSUE TITLE
Implement telemetry

### ISSUE THREAD

**Issue by luishfonseca** [GameDevTecnico/cubos#620]
# Checklist

- [ ] Remove spdlog
- [ ] Implement logging
- [ ] Implement metrics
- [ ] Implement tracing

# Problem
We aren't using any spdlog specific features, and it's a problematic dependency (#619), we should remove it.
As we are working on this, would also be good to expand telemetry to metrics and tracing.

# Design

- Superset of current logging interface
- Lightweight
- Consumable by 3rd party tooling

**Comment by RiscadoA** [comment]
Yeah that's what I had in mind, some kind of system where you could define scopes, and then you could filter log messages and metrics from it, perhaps? But yeah, I'm not really sure what we need here, maybe @luishfonseca has some clearer idea.

**Comment by luishfonseca** [comment]
Tbh I don't really remember what was the plan, I'm actually not that used to tracing in single machine applications, what does it achieve that we can't just do with logs and metrics?
